pkg/hub/rbacfinalizerdeletion/controller.go

```diff
@@ -44,76 +43,78 @@ type finalizeController struct {
 // NewFinalizeController ensures all manifestworks are deleted before role/rolebinding for work
 // agent are deleted in a terminating cluster namespace.
 func NewFinalizeController(
-	roleInformer rbacv1informers.RoleInformer,
-	roleBindingInformer rbacv1informers.RoleBindingInformer,
-	namespaceLister corelisters.NamespaceLister,
-	clusterLister clusterv1listers.ManagedClusterLister,
+	roleLister rbacv1listers.RoleLister,
+	roleBindingLister rbacv1listers.RoleBindingLister,
+	namespaceInformer corev1informers.NamespaceInformer,
+	clusterInformer informerv1.ManagedClusterInformer,
 	manifestWorkLister worklister.ManifestWorkLister,
 	rbacClient rbacv1client.RbacV1Interface,
 	eventRecorder events.Recorder,
 ) factory.Controller {
 
 	controller := &finalizeController{
-		roleLister:         roleInformer.Lister(),
-		roleBindingLister:  roleBindingInformer.Lister(),
-		namespaceLister:    namespaceLister,
-		clusterLister:      clusterLister,
+		roleLister:         roleLister,
+		roleBindingLister:  roleBindingLister,
+		namespaceLister:    namespaceInformer.Lister(),
+		clusterLister:      clusterInformer.Lister(),
 		manifestWorkLister: manifestWorkLister,
 		rbacClient:         rbacClient,
 		eventRecorder:      eventRecorder,
 	}
 
 	return factory.New().
 		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
+			accessor, _ := meta.Accessor(obj)
+			if accessor.GetDeletionTimestamp().IsZero() {
+				return ""
+			}
 			key, _ := cache.MetaNamespaceKeyFunc(obj)
 			return key
-		}, roleInformer.Informer(), roleBindingInformer.Informer()).
+		}, clusterInformer.Informer()).
+		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
+			accessor, _ := meta.Accessor(obj)
+			if accessor.GetDeletionTimestamp().IsZero() {
+				return ""
+			}
+			if _, ok := accessor.GetLabels()[clusterLabelKey]; !ok {
```

Comment:
if we filter the cluster label in the informer, we do not need this

Reply:
done